api/machinemanager/machinemanager.go

```diff
@@ -270,3 +270,26 @@ func (client *Client) GetUpgradeSeriesMessages(machineName, watcherId string) ([
 
 	return result.Result, nil
 }
+
+func (client *Client) Applications(machineName string) ([]string, error) {
+	if client.BestAPIVersion() < 5 {
+		return nil, errors.NotSupportedf("Applications")
+	}
+	args := params.Entities{Entities: []params.Entity{
+		{Tag: names.NewMachineTag(machineName).String()},
+	}}
+	results := new(params.StringsResults)
+	err := client.facade.FacadeCall("Applications", args, results)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if len(results.Results) != 1 {
+		return nil, errors.Errorf("expected 1 result, got %d", len(results.Results))
+	}
+	result := results.Results[0]
+	if result.Error != nil {
+		return nil, result.Error
+	}
+
+	return result.Result, nil
```

Comment:
It does make me wonder if it would be easier to read it as following (no change required, just musing)

```go
result := results.Results[0]
return result.Result, result.Error
```

Reply:
You need to check like this for API results, because the error member is actually a pointer.